ql/src/java/org/apache/hadoop/hive/ql/exec/repl/ReplDumpTask.java

```diff
@@ -146,7 +146,7 @@ public int execute() {
         }
         prepareReturnValues(Arrays.asList(currentDumpPath.toUri().toString(), String.valueOf(lastReplId)));
         writeDumpCompleteAck(hiveDumpRoot);
-        deletePreviousDumpMeta(previousDumpMetaPath);
+        deleteAllPreviousDumpMeta(dumpRoot, currentDumpPath);
```

Comment:
As load and dump are operating on same path ..how will it behave when load is reading these dump directory while dump is reading it ?

Reply:
Load should read only the latest one.

Reply:
But load ..reads all the path and loads only the latest one ..so there is a race condition ..

Reply:
status is a cached copy. So there won't be a problem